### PR TITLE
Notification Mitigation - AppNotificationManager::IsSupported

### PR DESCRIPTION
**Pull request by pmpurifoy**
Issue:
In elevated scenarios, PushNotifications do not work for both unpackaged + packaged applications and the IsSupported API doesn't include this limitation. Also for AppNotifications, there is no IsSupported API. Elevated scenarios are currently not supported for AppNotifications so we are added an IsElevated check to this newly added IsSupported API.

Fix:
Added IsElevated check to the PushNotification/AppNotification IsSupported API.

Verified:
Changes were tested by manually running scenarios of unpackaged/packaged + non-elevated/elevated and confirming correct behavior of AppNotification APIs.

### CODE REVIEW

File: dev/PushNotifications/PushNotificationManager.cpp [hunk 1]
```diff
@@ -130,7 +130,7 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
     bool PushNotificationManager::IsSupported()
     {
         // Only scenarios that use the Background Infrastructure component of the OS support Push in the SelfContained case
```
Comment (hulumane) [inline dev/PushNotifications/PushNotificationManager.cpp:132]: Update the comment to:
Packaged App scenario always works
[4:25 PM] Pavan Hullumane
Unpackaged App scenario does not

[4:25 PM] Pavan Hullumane
Unpackaged does not when it is self-contained or it is elevated

File: dev/PushNotifications/PushNotificationUtility.h [hunk 2]
```diff
@@ -143,6 +143,16 @@ namespace winrt::Microsoft::Windows::PushNotifications::Helpers
...
     }
 
+    inline bool IsElevated()
+    {
+        wil::unique_handle processToken;
+        THROW_IF_WIN32_BOOL_FALSE(OpenProcessToken(GetCurrentProcess(), MAXIMUM_ALLOWED, &processToken));
+
+        auto tokenLabel{ wil::get_token_information<TOKEN_MANDATORY_LABEL>(processToken.get()) };
+        DWORD subAuthority = static_cast<DWORD>(*GetSidSubAuthorityCount(tokenLabel->Label.Sid) - 1);
+        return *GetSidSubAuthority(tokenLabel->Label.Sid, subAuthority) >= SECURITY_MANDATORY_HIGH_RID;
```
Comment (hulumane) [inline dev/PushNotifications/PushNotificationUtility.h:153]: For elevated, can you add a comment that this equates to HIGH_RID? SYSTEM is not applicable right for elevated?

Comment (pmpurifoy) [reply dev/PushNotifications/PushNotificationUtility.h:153]: This just blocks all elevated + system callers. I can just use == instead. If that is preferred.

**Pull request closed without merging**